### PR TITLE
feat(notebook-controller): add readiness probe

### PR DESCRIPTION
**Pull request by jumana-s**
Adds a readiness probe to all notebooks without one.

Closes Statcan/daaas#339

### CODE REVIEW

File: components/notebook-controller/controllers/notebook_controller.go [hunk 1]
```diff
@@ -350,6 +350,18 @@ func generateStatefulSet(instance *v1beta1.Notebook) *appsv1.StatefulSet {
 		Value: "/notebook/" + instance.Namespace + "/" + instance.Name,
 	})
 
+	// Add readiness probe
+	if container.ReadinessProbe == nil {
```
Comment (cboin1996) [inline components/notebook-controller/controllers/notebook_controller.go:354]: This is just to prevent overwriting an existing probe right?

**Review by cboin1996 — COMMENTED** [review]
Looks good overall, just have one question out of curiosity.

File: components/notebook-controller/controllers/notebook_controller.go (hunk 1 above)
Comment (jumana-s) [reply components/notebook-controller/controllers/notebook_controller.go:354]: Yup, it was especially useful for testing.  Most if not all users create notebooks through the UI. 
If they created a notebook manually and added a readiness probe, they probably know what they're doing.

Comment (cboin1996) [reply components/notebook-controller/controllers/notebook_controller.go:354]: Ok great. Thanks for the explanation.

**Review by cboin1996 — APPROVED** [review]
LGTM

**Pull request merged**